airbyte-integrations/bases/bases-destination-jdbc/src/main/java/io/airbyte/integrations/destination/staging/GeneralStagingFunctions.java

```diff
@@ -77,8 +79,23 @@ public static void copyIntoTableFromStage(final JdbcDatabase database,
                                             final TyperDeduper typerDeduper)
       throws Exception {
     try {
-      stagingOperations.copyIntoTableFromStage(database, stageName, stagingPath, stagedFiles,
-          tableName, schemaName);
+      final Lock rawTableInsertLock = typerDeduper.getRawTableInsertLock(streamNamespace, streamName);
+      rawTableInsertLock.lock();
+      try {
+        stagingOperations.copyIntoTableFromStage(database, stageName, stagingPath, stagedFiles,
+            tableName, schemaName);
+      } finally {
+        rawTableInsertLock.unlock();
+      }
+
+      final AirbyteStreamNameNamespacePair streamId = new AirbyteStreamNameNamespacePair(streamName, streamNamespace);
+      if (!typerDeduperValve.containsKey(streamId)) {
+        typerDeduperValve.addStream(streamId);
+      }
```

Comment:
Maybe `putIfAbsent` is better here to avoid races

Reply:
oh that's a good call, though I think the valve is very nonthreadsafe anyway

Reply:
done in https://github.com/airbytehq/airbyte/pull/29878/commits/9270ab670d933f84f38eeb0ce51bf411fa1c4e6e